homeassistant/components/broadlink/light.py

```diff
@@ -110,7 +110,7 @@ async def async_turn_on(self, **kwargs: Any) -> None:
 
         elif ATTR_COLOR_TEMP in kwargs:
             color_temp = kwargs[ATTR_COLOR_TEMP]
-            state["colortemp"] = (color_temp - 153) * 100 + 2700
+            state["colortemp"] = round((color_temp - 2700) / 100 + 153)
```

Comment:
Looking at the library, I wonder if the bulb actually support color temperature in K and this is an attempt to convert from mired to K. @felipediel do you know it this is the case?

Can you try changing like this instead:
```diff
diff --git a/homeassistant/components/broadlink/light.py b/homeassistant/components/broadlink/light.py
index 796698c6a4c..bbf37502e30 100644
--- a/homeassistant/components/broadlink/light.py
+++ b/homeassistant/components/broadlink/light.py
@@ -7,7 +7,7 @@ from broadlink.exceptions import BroadlinkException
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_COLOR_MODE,
-    ATTR_COLOR_TEMP,
+    ATTR_COLOR_TEMP_KELVIN,
     ATTR_HS_COLOR,
     ColorMode,
     LightEntity,
@@ -80,7 +80,7 @@ class BroadlinkLight(BroadlinkEntity, LightEntity):
             self._attr_hs_color = [data["hue"], data["saturation"]]

         if "colortemp" in data:
-            self._attr_color_temp = round((data["colortemp"] - 2700) / 100 + 153)
+            self._attr_color_temp_kelvin = data["colortemp"]

         if "bulb_colormode" in data:
             if data["bulb_colormode"] == BROADLINK_COLOR_MODE_RGB:
@@ -108,9 +108,9 @@ class BroadlinkLight(BroadlinkEntity, LightEntity):
             state["saturation"] = int(hs_color[1])
             state["bulb_colormode"] = BROADLINK_COLOR_MODE_RGB

-        elif ATTR_COLOR_TEMP in kwargs:
-            color_temp = kwargs[ATTR_COLOR_TEMP]
-            state["colortemp"] = (color_temp - 153) * 100 + 2700
+        elif ATTR_COLOR_TEMP_KELVIN in kwargs:
+            color_temp = kwargs[ATTR_COLOR_TEMP_KELVIN]
+            state["colortemp"] = color_temp
             state["bulb_colormode"] = BROADLINK_COLOR_MODE_WHITE

         elif ATTR_COLOR_MODE in kwargs:
```

Reply:
Hi @emontnemery. Yes. Have a look [here](https://github.com/home-assistant/core/pull/73255/files), this implementation looks cleaner, but the PR was closed.